async_service/asyncio.py

```diff
@@ -290,6 +291,16 @@ def run_child_service(
         self.run_task(child_manager.run, daemon=daemon, name=task_name)
         return child_manager
 
+    @property
+    def stats(self) -> Stats:
+        total_count = max(0, len(self._service_task_dag) - 1)
+        finished_count = min(
+            total_count, len([task for task in self._service_task_dag if task.done()])
+        )
```

Comment:
Why bother with this `min()` check? Except for the weird case where all tasks are done, it should be the same number without the added constraint, right? (And why would we want that edge case to trigger, anyway?)

In other words, is this sufficient?
```py
finished_count = len([task for task in self._service_task_dag if task.done()])
```
If not, a tiny comment here could help readability.

Reply:
The `Service.run` method's "task" is part of the DAG but we don't want it to be included in the stats.  Without the `min/max` calls then we'd have an off-by-one error since it would be included in these counts.  We can't do the thing you posted since we can't easily discern which of those is the `run` method.

The `max` in the first ensures that we don't have `-1` tasks in some weird edge case before the `run` method has actually started.

the `min` call ensures that we don't count the `Service.run` method in the `finished_count` but it really only takes care of the case where **all** tasks are done.  There's an imperfect version of this where the `Service.run` method is done and the others are not and so the count isn't exactly right but I since it's complicated to handle now with the current APIs I'm choosing to address this in a subsequent PR when the task lifecycle has been more strictly defined.

Reply:
> There's an imperfect version of this where the `Service.run` method is done and the others are not and so the count isn't exactly right

So I guess this is what I would add in the comment. Something like: "The finished count is off-by-one, except when all tasks are finished, which is handled by `min()`." :+1: to just note it and move on.